react/Product.d.ts

```diff
@@ -0,0 +1,50 @@
+export function parseToJsonLD({
+  product,
+  selectedItem,
+  currency,
+  decimals,
+  pricesWithTax,
+}: {
+  product: any
+  selectedItem: any
+  currency: any
+  decimals: any
+  pricesWithTax: any
+}): {
+  '@context': string
+  '@type': string
+  '@id': string
+  name: any
+  brand: {
+    '@type': string
+    name: any
+  }
+  image: any
+  description: any
+  mpn: any
+  sku: any
+  category: any
+  offers: {
+    '@type': string
+    lowPrice: any
+    highPrice: any
+    priceCurrency: any
+    offers: any
+    offerCount: any
+  }
+}
+declare const _default: import('react').MemoExoticComponent<typeof StructuredData>
+export default _default
+declare function StructuredData({
+  product,
+  selectedItem,
+}: {
+  product: any
+  selectedItem: any
+}): JSX.Element
+declare namespace StructuredData {
+  namespace propTypes {
+    const product: any
+    const selectedItem: any
```

Comment:
Why did you use `any` for all these types? Is it possible to type them properly or use `unknown`?

Reply:
This was necessary to import the existing `Product.js` into `ProductList.tsx` ... this file was generated automatically via the `tsc` command and I guess the `any`s are due to the fact that the JS file by definition has no typings. Ideally `Product.js` would be rewritten using TypeScript, but I didn't feel qualified to undertake that effort.

Reply:
I've updated the file to use proper types where possible